app-engine/java/gradle/src/com/google/cloud/tools/intellij/appengine/java/gradle/AppEngineGradleModel.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.java.gradle;
+
+/** Model containing information obtained from the Gradle integration. */
+public interface AppEngineGradleModel {
+
+  /**
+   * Returns {@code true} if the module has the app-gradle-plugin configured, and {@code false}
+   * otherwise.
+   */
+  boolean hasAppEngineGradlePlugin();
+
+  /** Returns the path to the Gradle build directory. */
+  String gradleBuildDir();
```

Comment:
Could we think about a bit more specific name for a build dir or maybe more javadoc - sounds slightly generic?

Reply:
hmm. build directory is the same terminology used by Gradle itself. It really is just the directory where the gradle build is located. Do you have any suggestions?

Reply:
I was thinking that Gradle build directory could be for any of the modules, for build output, for build sources, just module root where build.gradle is? I guess that's the latter, correct, directory where build.gradle for the module is?

Reply:
so the AppEngineGradleModel is scoped to a single module, for one.

Also the `gradleBuildDir` is the build output directory under `../build` where the exploded war directory is located

Reply:
and to elaborate one why I'm storing this: its simply so that we can later get access to the exploded war directory (for gcloud deployment). This usage will come in the next PR or two.

Reply:
Thanks, now it's clear! we can probably add the last sentence into javadoc, wdyt?

Reply:
I'll add a sentence explaining what the build dir is: the build output where the built artifacts live. I'm hesitant to put anything about how it will be consumed since the model is agnostic of that.